minidump-processor/src/stackwalker/x86.rs

```diff
@@ -1,36 +1,52 @@
 // Copyright 2015 Ted Mielczarek. See the COPYRIGHT
 // file at the top-level directory of this distribution.
 
+// Note since x86 and Amd64 have basically the same ABI, this implementation
+// is written to largely erase the details of the two wherever possible,
+// so that it can be copied between the two with minimal changes. It's not
+// worth the effort to *actually* unify the implementations.
+
 use crate::process_state::{FrameTrust, StackFrame};
 use crate::stackwalker::unwind::Unwind;
 use minidump::format::CONTEXT_X86;
-use minidump::{MinidumpContext, MinidumpContextValidity, MinidumpMemory, MinidumpRawContext};
+use minidump::{
+    MinidumpContext, MinidumpContextValidity, MinidumpMemory, MinidumpModuleList,
+    MinidumpRawContext,
+};
 use std::collections::HashSet;
 
+type Pointer = u32;
+const POINTER_WIDTH: Pointer = 4;
+const INSTRUCTION_REGISTER: &str = "eip";
+const STACK_POINTER_REGISTER: &str = "esp";
+const FRAME_POINTER_REGISTER: &str = "ebp";
+
 fn get_caller_by_frame_pointer(
     ctx: &CONTEXT_X86,
     valid: &MinidumpContextValidity,
+    _trust: FrameTrust,
     stack_memory: &MinidumpMemory,
+    _modules: &MinidumpModuleList,
 ) -> Option<StackFrame> {
-    match *valid {
+    match valid {
         MinidumpContextValidity::All => {}
         MinidumpContextValidity::Some(ref which) => {
-            if !which.contains("ebp") {
+            if !which.contains(FRAME_POINTER_REGISTER) {
```

Comment:
I've had thoughts about some nicer abstraction here, like a bitset or enum set of valid registers, or maybe even a type that wraps a raw context like `Partial<CONTEXT_X86>` that would give you accessor functions for each register that return `Option<T>`.  Certainly not critical, but the latter would make code that handles contexts more resilient against using invalid register contents.

(Just writing down thoughts that reading this dredged up from my memory, nothing you need to address at the moment!)

Reply:
My first pass at this that was inside the core Thread reading did introduce some steps toward more abstractions and a platform-agnostic unwinding, but i went for the path of least resistance here and just used platform-specific stuff that's copy-pastable. 

WIP hacky old version:
https://github.com/luser/rust-minidump/commit/62638ac37f5bb0af3b025edaa7e70d7a18ee3d19#diff-d6346fd7d17270b1282142aeeda9c4bc2b7d8fd0f37b24a1c871a9257f0ed0aaR398-R429